tornado-examples/src/main/java/uk/ac/manchester/tornado/examples/compute/MatrixVectorRowMajor.java

```diff
@@ -0,0 +1,337 @@
+        System.out.println("======================================");
+
+        // Default parameters
+        int inputDim = 8192;   // Default input dimension (columns)
+        int outputDim = 2048; // Default output dimension (rows)
+
+        // Parse command line arguments if provided
+        if (args.length >= 3) {
+            try {
+                inputDim = Integer.parseInt(args[0]);
+                outputDim = Integer.parseInt(args[1]);
+                int localWorkGroupSize = Integer.parseInt(args[2]);
+            } catch (NumberFormatException e) {
+                System.err.println("Error parsing dimensions. Using defaults.");
+            }
+        }
+
+        System.out.println("Configuration:");
+        System.out.println("- Input dimension (columns): " + inputDim);
+        System.out.println("- Output dimension (rows): " + outputDim);
+        System.out.println("- Local work group size: " + LOCAL_WORK_GROUP_SIZE);
+        System.out.println("- Warmup iterations: " + WARM_UP_ITERATIONS);
+        System.out.println("- Benchmark iterations: " + BENCHMARK_ITERATIONS);
+        System.out.println();
+
+        // Create data arrays
+        FloatArray input = new FloatArray(inputDim);
+        FloatArray weights = new FloatArray(inputDim * outputDim);
+        FloatArray outputParallel = new FloatArray(outputDim);
+        FloatArray outputPureTornado = new FloatArray(outputDim);
+        FloatArray outputSeq = new FloatArray(outputDim);
+
+        // Initialize data
+        System.out.println("Initializing data...");
+        fillRandomData(input, -1.0f, 1.0f);
+        fillRandomData(weights, -0.1f, 0.1f);
+
+        // Arrays for timing measurements
+        ArrayList<Long> sequentialTimers = new ArrayList<>();
+        ArrayList<Long> parallelTimers = new ArrayList<>();
+        ArrayList<Long> pureTornadoTimers = new ArrayList<>();
+
+        // Set up TornadoVM execution
+        System.out.println("Setting up TornadoVM execution...");
+        WorkerGrid1D worker = new WorkerGrid1D(outputDim * LOCAL_WORK_GROUP_SIZE);
+        GridScheduler scheduler = new GridScheduler("s0.t0", worker);
+        worker.setLocalWork(LOCAL_WORK_GROUP_SIZE, 1, 1);
+
+        TaskGraph taskGraph = new TaskGraph("s0").transferToDevice(DataTransferMode.FIRST_EXECUTION, input, weights)
+                .task("t0", MatrixVectorRowMajor::matrixVectorGeneric, new KernelContext(), input, outputParallel, weights, inputDim, outputDim, LOCAL_WORK_GROUP_SIZE)
+                .transferToHost(DataTransferMode.EVERY_EXECUTION, outputParallel);
+
+        ImmutableTaskGraph immutableTaskGraph = taskGraph.snapshot();
+
+        TaskGraph taskGraphPure = new TaskGraph("s1").transferToDevice(DataTransferMode.FIRST_EXECUTION, input, weights)
+                .task("t0", MatrixVectorRowMajor::matrixVectorParallel, input, outputPureTornado, weights, inputDim, outputDim).transferToHost(DataTransferMode.EVERY_EXECUTION, outputPureTornado);
```

Comment:
can you unfold this `.transferToHost(DataTransferMode.EVERY_EXECUTION, outputPureTornado);` into next line?